benchmarks/java/src/main/java/com/google/protobuf/ProtoBench.java

```diff
@@ -94,19 +107,69 @@ public static boolean runTest(String file) {
     }
     argsList.add("-DdataFile=" + file);
     argsList.add("com.google.protobuf.ProtoBenchCaliper");
+    String[] caliperArgs = new String[argsList.size() + Math.max(args.length - endIndex - 1, 0)];
+    argsList.toArray(caliperArgs);
+    if (endIndex != args.length) {
+      System.arraycopy(args, endIndex + 1, 
+          caliperArgs, argsList.size(), args.length - endIndex - 1);
+    }
     
     try {
-      String args[] = new String[argsList.size()];
-      argsList.toArray(args);
-      CaliperMain.exitlessMain(args, 
+      CaliperMain.exitlessMain(caliperArgs, 
           new PrintWriter(System.out, true), new PrintWriter(System.err, true));
-      return true;
     } catch (Exception e) {
       System.err.println("Error: " + e.getMessage());
       System.err.println("Detailed exception information:");
       e.printStackTrace(System.err);
       return false;
     }
+    try {
+      double maxTimeIntervalScale = 0;
+      Scanner scanner = new Scanner(new String(readAllBytes("JavaBenchmarkWarning.txt")));
+      while (scanner.hasNext()) {
+        maxTimeIntervalScale = Math.max(maxTimeIntervalScale, scanner.nextDouble());
+      }
+      scanner.close();
+
+      String timingIntervalWithUnit = getFromArgs(caliperArgs, 
+          "instrument.runtime.options.timingInterval");
+      if (timingIntervalWithUnit.equals("")) {
+        timingIntervalWithUnit = "500ms";
+      }
+      int doubleValueEndIndex = timingIntervalWithUnit.length();
+      double timingIntervalValue = 0;
+      for (int i = timingIntervalWithUnit.length(); i >= 0 ; i--) {
+        try {
+          timingIntervalValue = Double.parseDouble(timingIntervalWithUnit.substring(0, i));
+          doubleValueEndIndex = i;
+          break;
+        } catch (NumberFormatException e) {
+          
+        } catch (NullPointerException e) {
+          
+        }
+      } 
+      timingIntervalValue *= maxTimeIntervalScale * 1.2;
+      String measurementsString = getFromArgs(caliperArgs, 
+          "instrument.runtime.options.measurements");
+      if (measurementsString.equals("")) {
+        measurementsString = "9";
+      }
+      int measurements = (int) (Integer.parseInt(measurementsString) * 1.2 * maxTimeIntervalScale);
+      
+      System.out.println(
```

Comment:
It seems this warning is printed unconditionally?

Reply:
If there's no file "JavaBenchmarkWarning.txt" then there will be IOException thrown inside try catch, and won't print the warning.